File: chap02/Ordinal.java
// in chap02 run >javac Ordinal.java
// and then >java Ordinal 3

public class Ordinal {
  
  public static void main(String[] args) {
    ordinal(Integer.parseInt(args[0]));
  }
  
  public static void ordinal(int number) {
    switch(number) {
      case 1: System.out.println("1st"); break;
      case 2: System.out.println("2nd"); break;
      case 3: System.out.println("3rd"); break;
      case 4: System.out.println("4th"); break;
      case 5: System.out.println("5th"); break;
      case 6: System.out.println("6th"); break;
      case 7: System.out.println("7th"); break;
      case 8: System.out.println("8th"); break;
      case 9: System.out.println("9th"); break;
      case 10: System.out.println("10th"); break;
      default : System.out.println("Cannot do beyond 10");
    }
  }
}

File: chap02/Ordinal.scala
// Run with e.g. >scala Ordinal.scala 9 
// in chap02/

ordinal(args(0).toInt)

def ordinal(number:Int) = number match {
  case 1 => println("1st")
  case 2 => println("2nd")
  case 3 => println("3rd")
  case 4 => println("4th")
  case 5 => println("5th")
  case 6 => println("6th")
  case 7 => println("7th")
  case 8 => println("8th")
  case 9 => println("9th")
  case 10 => println("10th")
  case _ => println("Cannot do beyond 10")
}